### PR TITLE
Nrunner: improvements to the status server

### PR DESCRIPTION
**Pull request by clebergnu**
The most important feature is a list of the failed/errored tasks, which currently usually requires a scrollback/search of the previous output.

### CODE REVIEW

**Comment by clebergnu** [comment]
@beraldoleal FIY, I've added two commits to this PR, please refresh your local copy if you have one.

File: avocado/core/nrunner.py [hunk 1]
```diff
@@ -727,30 +727,30 @@ def handle_task_started(self, data):
...
-                self.result[data['result']] += 1
-            else:
-                self.result[data['result']] = 1
-
-        if data.get('result') not in ('pass', 'skip'):
+        if 'result' not in data:
+            return
+
+        result = data['result']
+        id_ = data['id']
```
Comment (beraldoleal) [inline avocado/core/nrunner.py:734]: I would like to suggest to be explicit here, what id ? `task_id`?

File: avocado/core/nrunner.py [hunk 2]
```diff
@@ -736,10 +736,9 @@ def handle_task_finished(self, data):
         self.tasks_pending.remove(id_)
         print('Task complete (%s): %s' % (result, id_))
 
-        if result in self.result:
-            self.result[result] += 1
-        else:
-            self.result[result] = 1
+        if result not in self.result:
+            self.result[result] = []
+        self.result[result].append(id_)
```
Comment (beraldoleal) [inline avocado/core/nrunner.py:741]: +1 for this! :+1:

File: avocado/plugins/nrun.py [hunk 3]
```diff
@@ -95,6 +95,16 @@ def spawn_tasks(self, parallel_tasks):
             else:
                 LOG_UI.info("%s spawned and alive", identifier)
 
+    def report_results(self):
```
Comment (beraldoleal) [inline avocado/plugins/nrun.py:102]: I might be wrong, but it looks like this is a `StatusServer` method? What is your option about moving this to `StatusServer` class?

File: avocado/plugins/nrun.py [hunk 4]
```diff
@@ -85,16 +85,30 @@ def spawn_tasks(self, parallel_tasks):
                 print("Finished spawning tasks")
                 break
 
-            yield from self.spawner.spawn_task(task)
+            spawn_result = yield from self.spawner.spawn_task(task)
             identifier = task.identifier
             self.pending_tasks.remove(task)
             self.spawned_tasks.append(identifier)
+            if not spawn_result:
```
Comment (beraldoleal) [inline avocado/plugins/nrun.py:92]: Just one small comment here. Maybe `spawner.spawn_task` could raise `SpawnerExceptions` with details about what was wrong, so here you could to a `try...except`.

File: avocado/plugins/nrun.py (hunk 3 above)
Comment (clebergnu) [reply avocado/plugins/nrun.py:102]: I understand your point, but on the other hand, I don't see the "presentation layer" as a responsibility of the `StatusServer` class.  I think that's more the responsibility `nrun` command/plugin.  The `StatusServer` is providing the data, though.

File: avocado/plugins/nrun.py (hunk 4 above)
Comment (clebergnu) [reply avocado/plugins/nrun.py:92]: It also concerns me that we don't have more detailed information about the failures, and I see that as a further improvement indeed.  I can try to include that in the next (or next next round), which should include scheduler keeping track of those details.  How does that sound?

**Pull request closed without merging**